Formula/rocketmq.rb

```diff
@@ -0,0 +1,133 @@
+class Rocketmq < Formula
+  desc "Distributed messaging and streaming platform"
+  homepage "https://rocketmq.apache.org"
+  url "https://www.apache.org/dyn/closer.lua?path=rocketmq/4.9.0/rocketmq-all-4.9.0-bin-release.zip"
+  mirror "https://archive.apache.org/dist/rocketmq/4.9.0/rocketmq-all-4.9.0-bin-release.zip"
+  sha256 "d13416f9e80f5d4414300fbf84518a22476e7e298d19c6b8d21c7e99c4599065"
+  license "Apache-2.0"
+
+  depends_on "openjdk"
+
+  resource "netty-all" do
+    url "https://search.maven.org/remotecontent?filepath=io/netty/netty-all/4.1.65.Final/netty-all-4.1.65.Final.jar"
+    sha256 "e6cdae2b23e410b35128dac651b954aedd583bc10959a1e702c7dba7ac850d0e"
+  end
+
+  def install
+    inreplace "bin/runserver.sh" do |s|
+      s.gsub! "export CLASSPATH=.:${BASE_DIR}/conf:${CLASSPATH}",
+        "export CLASSPATH=.:${BASE_DIR}/lib/*:${BASE_DIR}/conf:${CLASSPATH}"
+      s.gsub! "JAVA_OPT=\"${JAVA_OPT} -Djava.ext.dirs", "#JAVA_OPT=\"${JAVA_OPT} -Djava.ext.dirs"
+      s.gsub! "choose_gc_log_directory\n", ""
+      s.gsub! "choose_gc_options\n", "\n"
+    end
+
+    inreplace "bin/runbroker.sh" do |s|
+      s.gsub! "export CLASSPATH=.:${BASE_DIR}/conf:${CLASSPATH}",
+        "export CLASSPATH=.:${BASE_DIR}/lib/*:${BASE_DIR}/conf:${CLASSPATH}"
+      s.gsub! "JAVA_OPT=\"${JAVA_OPT} -Djava.ext.dirs", "#JAVA_OPT=\"${JAVA_OPT} -Djava.ext.dirs"
+      s.gsub! "JAVA_OPT=\"${JAVA_OPT} -verbose:gc -Xloggc:${GC_LOG_DIR}/rmq_broker_gc_%p_%t.log -XX:+PrintGCDetails" \
+        " -XX:+PrintGCDateStamps -XX:+PrintGCApplicationStoppedTime -XX:+PrintAdaptiveSizePolicy\"", ""
+      s.gsub! "JAVA_OPT=\"${JAVA_OPT} -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=5 -XX:GCLogFileSize=30m\"", ""
+      s.gsub! "JAVA_OPT=\"${JAVA_OPT} -XX:-UseLargePages -XX:-UseBiasedLocking\"",
+        "JAVA_OPT=\"${JAVA_OPT} -XX:-UseLargePages\""
+      s.gsub! "choose_gc_log_directory\n", "\n"
+    end
+
+    inreplace "bin/tools.sh" do |s|
+      s.gsub! "export CLASSPATH=.:${BASE_DIR}/conf:${CLASSPATH}",
+        "export CLASSPATH=.:${BASE_DIR}/lib/*:${BASE_DIR}/conf:${CLASSPATH}"
+      s.gsub! "JAVA_OPT=\"${JAVA_OPT} -Djava.ext.dirs", "#JAVA_OPT=\"${JAVA_OPT} -Djava.ext.dirs"
+    end
```

Comment:
New formulae should not require patches to work. Please submit these fixes upstream.

https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff